internal/gatewayapi/route.go

```diff
@@ -18,8 +18,8 @@ import (
 
 var (
 	_                RoutesTranslator = (*Translator)(nil)
-	validServiceName                  = `^(?i)\.?[a-z_][a-z_0-9]*(\.[a-z_][a-z_0-9]*)*$`
-	validMethodName                   = `^[A-Za-z_][A-Za-z_0-9]*$`
+	validServiceName                  = `(?i)\.?[a-z_][a-z_0-9]*(\.[a-z_][a-z_0-9]*)*`
```

Comment:
curious why the `^` and `$` was removed

Reply:
cause if we do not remove `^` and `$`, the final regex pattern will be like `/^...$/method`, and it does not match any request.

Reply:
Thanks for the clarification